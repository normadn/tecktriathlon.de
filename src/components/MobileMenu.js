import React from "react";

import cubeswebp from "../assets/img/icons/mobileMenuCubes.webp";
import menuClose from "../assets/img/icons/mobileMenuClose.svg";
import hamburger from "../assets/img/icons/mobileMenuOpen.svg";
import word from "../assets/img/logo/amsel-tech-word-white.svg";
import cubespng from "../assets/img/icons/mobileMenuCubes.png";
import logo from "../assets/img/logo/amsel-logo-bird-small-white-new.svg";

const MobileMenu = (props) => (
  <div
    align="right"
    css={{
      marginLeft: "auto",
      paddingTop: "32px",
      "@media(min-width: 820px)": {
        display: "none"
      }
    }}
  >
    <input
      type="image"
      onClick={props.toggleMenu}
      src={hamburger}
      css={{ width: "32px", height: "24px" }}
      alt="Open menu"
    />
    <div
      className={props.isMenuVisible ? "no-scroll" : ""}
      css={{
        display: props.isMenuVisible ? "block" : "none",
        position: "fixed",
        width: "100vw",
        height: "100vh",
        overflow: "hidden",
        top: "0px",
        left: "0px",
        zIndex: "5",
        background: "#11213f"
      }}
    >
      <input
        type="image"
        align="right"
        css={{ padding: "32px" }}
        src={menuClose}
        onClick={props.toggleMenu}
        alt="Close menu"
      />
      <div
        css={{
          marginTop: "100px",
          color: "white",
          display: "grid",
          gridTemplateColumns: "1fr"
        }}
      >
        <div
          css={{
            marginLeft: "auto",
            marginRight: "auto",
            textAlign: "center",
            " a": {
              color: "white",
              textTransform: "uppercase",
              ":hover": {
                color: "#fedb01"
              }
            }
          }}
        >
          <div>
            <img
              src={logo}
              css={{ paddingBottom: "11px", width: "45%"}}
              alt="QHacks Crown"
            />
          </div>
          <div css={{ paddingBottom: "52px" }}>
            <img
              src={word}
              width="134px"
              height="34.8px"
              alt="QHacks Wordmark"
            />
          </div>
          {props.menuItems.map((i) => (
            <div key={i} css={{ paddingBottom: "24px" }}>
              <a
                href={`#${i}`}
                onClick={props.toggleMenu}
                css={{ fontWeight: "bold" }}
              >
                {i}
              </a>
            </div>
          ))}
          <div css={{ marginTop: "-100px", pointerEvents: "none" }}>
            <picture>
              <source srcSet={cubeswebp} type="image/webp" />
              <source srcSet={cubespng} type="image/png" />
              <img src={cubespng} alt="Floating cubes" />
            </picture>
          </div>
        </div>
      </div>
    </div>
  </div>
);

export default MobileMenu;
