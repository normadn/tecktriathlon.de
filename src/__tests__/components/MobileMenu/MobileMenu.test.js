import React from "react";
import MobileMenu from "../../../components/MobileMenu";
import { shallow } from "enzyme";

describe("MobileMenu", () => {
  it("renders correctly", () => {
    expect(
      shallow(<MobileMenu menuItems={["I", "LOVE", "SnApShOtS"]} />)
    ).toMatchSnapshot();
  });
});
