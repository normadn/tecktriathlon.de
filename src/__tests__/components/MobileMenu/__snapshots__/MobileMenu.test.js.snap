// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileMenu renders correctly 1`] = `
<div
  align="right"
  css={
    Object {
      "@media(min-width: 820px)": Object {
        "display": "none",
      },
      "marginLeft": "auto",
      "paddingTop": "32px",
    }
  }
>
  <input
    alt="Open menu"
    css={
      Object {
        "height": "24px",
        "width": "32px",
      }
    }
    src="test-file-stub"
    type="image"
  />
  <div
    className=""
    css={
      Object {
        "background": "#11213f",
        "display": "none",
        "height": "100vh",
        "left": "0px",
        "overflow": "hidden",
        "position": "fixed",
        "top": "0px",
        "width": "100vw",
        "zIndex": "5",
      }
    }
  >
    <input
      align="right"
      alt="Close menu"
      css={
        Object {
          "padding": "32px",
        }
      }
      src="test-file-stub"
      type="image"
    />
    <div
      css={
        Object {
          "color": "white",
          "display": "grid",
          "gridTemplateColumns": "1fr",
          "marginTop": "100px",
        }
      }
    >
      <div
        css={
          Object {
            " a": Object {
              ":hover": Object {
                "color": "#fedb01",
              },
              "color": "white",
              "textTransform": "uppercase",
            },
            "marginLeft": "auto",
            "marginRight": "auto",
            "textAlign": "center",
          }
        }
      >
        <div>
          <img
            alt="QHacks Crown"
            css={
              Object {
                "paddingBottom": "11px",
              }
            }
            src="test-file-stub"
          />
        </div>
        <div
          css={
            Object {
              "paddingBottom": "52px",
            }
          }
        >
          <img
            alt="QHacks Wordmark"
            height="34.8px"
            src="test-file-stub"
            width="134px"
          />
        </div>
        <div
          css={
            Object {
              "paddingBottom": "24px",
            }
          }
          key="I"
        >
          <a
            css={
              Object {
                "fontWeight": "bold",
              }
            }
            href="#I"
          >
            I
          </a>
        </div>
        <div
          css={
            Object {
              "paddingBottom": "24px",
            }
          }
          key="LOVE"
        >
          <a
            css={
              Object {
                "fontWeight": "bold",
              }
            }
            href="#LOVE"
          >
            LOVE
          </a>
        </div>
        <div
          css={
            Object {
              "paddingBottom": "24px",
            }
          }
          key="SnApShOtS"
        >
          <a
            css={
              Object {
                "fontWeight": "bold",
              }
            }
            href="#SnApShOtS"
          >
            SnApShOtS
          </a>
        </div>
        <div
          css={
            Object {
              "marginTop": "-100px",
              "pointerEvents": "none",
            }
          }
        >
          <picture>
            <source
              srcSet="test-file-stub"
              type="image/webp"
            />
            <source
              srcSet="test-file-stub"
              type="image/png"
            />
            <img
              alt="Floating cubes"
              src="test-file-stub"
            />
          </picture>
        </div>
      </div>
    </div>
  </div>
</div>
`;
